sklearn/cluster/_kmeans.py

```diff
@@ -275,15 +275,51 @@ class KMeansCythonEngine:
     TODO: see URL for more details.
     """
 
+    @staticmethod
+    def convert_to_sklearn_types(name, value):
+        """Convert estimator attributes to scikit-learn types.
+
+        Users can configure whether estimator attributes should be stored
+        using engine native types or scikit-learn types. This function is
+        used to convert attributes from engine to scikit-learn native types.
+
+        Scikit-learn native types are ndarrays and basic Python types. There
+        is no need to convert these.
+
+        Parameters
+        ----------
+        name : str
+            Name of the attribute being converted.
+
+        value
+            Value of the attribute being converted.
+
+        Returns
+        --------
+        converted
+            Attribute value converted to a scikit-learn native type.
+        """
+        # XXX Maybe a bit useless as it should never get called, but it
+        # does demonstrate the API
+        return value
+
     def __init__(self, estimator):
         self.estimator = estimator
 
     def accepts(self, X, y=None, sample_weight=None):
+        """Validate input and hyper-parameters.
+
+        Determine if the engine can handle the hyper-parameter of the estimator
+        as well as the input data.
```

Comment:
```suggestion
        as well as the input data. If not, this would allow to select another
        engine positioned after in the current provider list or ultimately
        the default engine of the estimator as fallback.
```

Reply:
I edited the docstring to incorporate what you suggested but in a slightly different way.

I feel like the explanation of how the engine selection works somehow should be somewhere else no? Somewhere that "defines" the engine interface. Maybe something to explain in more detail in the documentation (I don't think we should introduce some kind of base engine class just to define the API, feels wrong.)